diff --git a/src/AbstractBubble.vala b/src/AbstractBubble.vala
index 197fe0b..4fc53c2 100644
--- a/src/AbstractBubble.vala
+++ b/src/AbstractBubble.vala
@@ -166,7 +166,9 @@ public class Notifications.AbstractBubble : Gtk.Window {
         }

         if (Gdk.Display.get_default () is Gdk.X11.Display) {
-            // Avoid present on X11 because it focuses the window
+            // Avoid present on X11 because it focuses the window
+            // Realize first so we can set the window type atom before the WM sees it
+            realize ();
             base.show ();
         } else {
             base.present ();
